src/mapping/tests/RadialGeoMultiscaleMappingTest.cpp

```diff
@@ -0,0 +1,124 @@
+  int     inDataID = inData->getID();
+  inMesh->createVertex(Eigen::Vector3d::Constant(0.0));
+  inMesh->createVertex(Eigen::Vector3d(3.0, 0.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(6.0, 0.0, 0.0));
+  inMesh->allocateDataValues();
+  Eigen::VectorXd &inValues = inData->values();
+  inValues << 1.0, 2.0, 3.0;
+
+  // Create mesh to map to
+  PtrMesh outMesh(new Mesh("OutMesh", dimensions, testing::nextMeshID()));
+  PtrData outData   = outMesh->createData("OutData", 1, 2_dataID);
+  int     outDataID = outData->getID();
+  outMesh->createVertex(Eigen::Vector3d(1.0, 1.0, 0.0)); // closest to first 1D vertex
+  outMesh->createVertex(Eigen::Vector3d(4.0, 2.0, 0.0)); // closest to second 1D vertex
+  outMesh->createVertex(Eigen::Vector3d(7.0, 3.0, 0.0)); // closest to third 1D vertex
+  outMesh->allocateDataValues();
+
+  // Setup mapping with mapping coordinates and geometry used
+  precice::mapping::RadialGeoMultiscaleMapping mapping(mapping::Mapping::CONSISTENT, dimensions, mapping::RadialGeoMultiscaleMapping::SPREAD, mapping::RadialGeoMultiscaleMapping::X);
+  mapping.setMeshes(inMesh, outMesh);
+  BOOST_TEST(mapping.hasComputedMapping() == false);
+
+  // Map data
+  mapping.computeMapping();
+  mapping.map(inDataID, outDataID);
+  const Eigen::VectorXd &outValues = outData->values();
+
+  // Check if data is mapped to closest vertex
+  BOOST_TEST(mapping.hasComputedMapping() == true);
+  BOOST_TEST(outValues(0) == inValues(0));
+  BOOST_TEST(outValues(1) == inValues(1));
+  BOOST_TEST(outValues(2) == inValues(2));
+}
+
+BOOST_AUTO_TEST_CASE(testConsistentCollect)
+{
+  /*  The following test works by creating two dimensionally heterogeneous meshes, namely 1D and 3D, that intersect along the x-axis.
+      Then, the data is mapped from the vertices of the 3D mesh to defined vertices of the 1D mesh (hence, COLLECT).
+      The defined vertices are close to a 1D vertex and therefore we can predict that the 1D vertex value should be the mean of all close 3D vertices.
+      Finally, this expected behavior is tested.  
+  */
+
+  PRECICE_TEST(1_rank);
+  int dimensions = 3;
+  using testing::equals;
+
+  // Create mesh to map from
+  PtrMesh inMesh(new Mesh("InMesh", dimensions, testing::nextMeshID()));
+  PtrData inData   = inMesh->createData("InData", 1, 0_dataID);
+  int     inDataID = inData->getID();
+  inMesh->createVertex(Eigen::Vector3d(1.0, 1.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(1.0, 2.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(2.0, 1.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(4.0, 1.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(5.0, 2.0, 0.0));
+  inMesh->createVertex(Eigen::Vector3d(7.0, 1.0, 0.0));
+  inMesh->allocateDataValues();
+  Eigen::VectorXd &inValues = inData->values();
+  inValues << 1.0, 3.0, 5.0, 7.0, 2.0, 4.0;
+
+  // Create mesh to map to
+  PtrMesh outMesh(new Mesh("OutMesh", dimensions, testing::nextMeshID()));
+  PtrData outData   = outMesh->createData("OutData", 1, 2_dataID);
```

Comment:
Instead of 1,2,3 for the dimensions argument, it would be much more readable if we had an enum for the number of dimensions. One could then write:

```c++
outMesh->createData("outData", DIMENSIONS::THREE, 2_dataID)
```

Reply:
Then that's fine, let's leave it like this.